kif-framework/KIF
# Added ability to enable/disable autocorrect/smart quotes/smart dashes

Resolves #1012

This disables autocorrect (as well as smart quotes/smart dashes for iOS 11+) by default, allowing the user to enable/disable them on a per-suite basis (at the minimum level of granularity). I've added some tests to ensure that this behavior is maintained.

This is done by swizzling `UITextField`'s `autocorrectionType`, `smartQuotesType` and `smartDashesType` properties to always return `UITextAutocorrectionTypeNo` `UITextSmartQuotesTypeNo`, and `UITextSmartDashesTypeNo`, respectively. This is entirely reversible, as well, as is demonstrated in the UI tests added.

**KIF Framework Changes:**
- Autocorrect, smart quotes and smart dashes are now disabled by default. In order to re-enable them, override the test suite's `+(void)setUp` method, and call `[KIFTestActor setEnableAutocorrect:YES]`, `[KIFTestActor setEnableSmartQuotes:YES]` or `[KIFTestActor setEnableSmartDashes:YES]`

**Test Changes:**
- Ensure that dashes are modified when smart dashes are enabled, as well as that they're not modified when smart dashes are disabled.
- Ensure that quotes are modified when smart quotes are enabled, as well as that they're not modified when smart quotes are disabled.
- Ensure that autocorrect works when it's enabled, and that it doesn't when it's disabled.
